### DISCUSSION TITLE
What is Done in Lesson 12? Explain the Flow with some example

### DISCUSSION THREAD

**Discussion by AshNewar** [Cyfrin/foundry-full-course-cu#852] [Q&A]
I could not understand What does DSCEngine does and  What are we doing in the code ?
Can any one explain the whole procedure in form of example or just tell me the flow 
Needed Help Please..

**Comment by 0xusmanf** [comment]
Do you have an understanding of how Defi works? And have you finished lesson 12?

Answers to the above questions will help me redirect you to the appropriate resources.

**Reply by AshNewar** [reply]
no i didn,t understand how defi works , and yes i have finished the lesson12

**Reply by 0xusmanf** [reply]
[This](https://teachyourselfcrypto.com/#ftoc-module-4-decentralized-finance-defi) is an excellent resource for understanding DeFi for beginners. Finish module 4 of this resource and you will not only understand what is happening in the DSCEngine.sol but you will understand how DeFi works as a whole.
